src/tests/providers/test_vultr.py

```diff
@@ -0,0 +1,1441 @@
+        memory=960.0,
+        gpu_count=16,
+        gpu_name="A16",
+        gpu_memory=256,
+        spot=False,
+        disk_size=1700,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vcg-a100-48c-480g-320vram",
+        location="ewr",
+        price=9.589,
+        cpu=48,
+        memory=480.0,
+        gpu_count=8,
+        gpu_name="A100",
+        gpu_memory=320,
+        spot=False,
+        disk_size=1400,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vbm-48c-1024gb-4-a100-gpu",
+        location="ewr",
+        price=9.59,
+        cpu=48,
+        memory=1024.0,
+        gpu_count=4,
+        gpu_name="A100",
+        gpu_memory=160,
+        spot=False,
+        disk_size=450,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vcg-l40s-128c-1500g-384vram",
+        location="ewr",
+        price=13.368,
+        cpu=128,
+        memory=1500.0,
+        gpu_count=8,
+        gpu_name="L40S",
+        gpu_memory=384,
+        spot=False,
+        disk_size=3400,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vbm-256c-2048gb-8-mi300x-gpu",
+        location="ord",
+        price=16.13,
+        cpu=128,
+        memory=2267.50390625,
+        gpu_count=8,
+        gpu_name="MI300X",
+        gpu_memory=1536,
+        spot=False,
+        disk_size=3576,
+        gpu_vendor="amd",
+    ),
+    RawCatalogItem(
+        instance_name="vbm-64c-2048gb-8-l40-gpu",
+        location="ewr",
+        price=16.44,
+        cpu=64,
+        memory=2048.0,
+        gpu_count=8,
+        gpu_name="L40",
+        gpu_memory=384,
+        spot=False,
+        disk_size=480,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vbm-112c-2048gb-8-a100-gpu",
+        location="atl",
+        price=20.62,
+        cpu=112,
+        memory=2048.0,
+        gpu_count=8,
+        gpu_name="A100",
+        gpu_memory=320,
+        spot=False,
+        disk_size=960,
+        gpu_vendor="nvidia",
+    ),
+    RawCatalogItem(
+        instance_name="vbm-112c-2048gb-8-h100-gpu",
+        location="sea",
+        price=22.02,
+        cpu=112,
+        memory=2048.0,
+        gpu_count=8,
+        gpu_name="H100",
+        gpu_memory=640,
+        spot=False,
+        disk_size=960,
+        gpu_vendor="nvidia",
+    ),
+]
+
+
+@pytest.fixture
+def mock_fetch_offers(mocker):
+    return mocker.patch("gpuhunt.providers.vultr.fetch_offers", return_value=machine_types)
+
+
+def test_vultr_provider_get(mock_fetch_offers):
+    catalog = Catalog(balance_resources=False, auto_reload=False)
+    vultr = VultrProvider()
+    internal_catalog.ONLINE_PROVIDERS = ["vultr"]
+    internal_catalog.OFFLINE_PROVIDERS = []
+    catalog.add_provider(vultr)
+    query_result = catalog.query(provider=["vultr"], min_gpu_count=2, max_gpu_count=4)
+    assert len(query_result) >= 1, "No offers found"
```

Comment:
This test only covers line 26 of `vultr.py`, the rest is mocked. I think it'd be much more useful to only mock the requests to `/plans` and `/plans-metal`, so that all the code in `vultr.py` is covered by the test.

It also seems quite redundant to have 1.5K lines of mocked data and then only check for `len(query_result) >= 1`. I can suggest reducing the size of the mocked data so that only the items relevant to the test case are included.